### PR TITLE
Case study assignment1

### PR DESCRIPTION
**Pull request by Harshitha-Somashekar**

### CODE REVIEW

**Review by pavangeeber — APPROVED** [review]
Doubt: converting numbers to str in print statements is a necessity or a good practice?

File: Factors.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+number=input("Enter the number:")
+factors = []
+for i in range(1,int(number)):
```
Comment (pavangeeber) [inline Factors.py]: loop does not cover the entered number, for i in range (1,int(number+1)) should fix it.

File: FindEvenDigitNumbers.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+num1 = 1000
+num2 = 3000
+numberList = []
+for i in range(num1, num2 + 1):
+    numberList.append(i)
+    for j in str(i):
```
Comment (pavangeeber) [inline FindEvenDigitNumbers.py:6]: super

File: FindNumDigitsLetters.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+sentence = input("Enter the sentence:")
+numOfDigits = 0
+numOfLetters = 0
+for i in sentence:
+    if i.isdigit():
+        numOfDigits += 1
+    elif i.isalpha():
```
Comment (pavangeeber) [inline FindNumDigitsLetters.py:7]: Great! efficient use of elif than just another if.

**Review by pavangeeber — APPROVED** [review]

**Comment by Harshitha-Somashekar** [comment]
> 
> 
> Doubt: converting numbers to str in print statements is a necessity or a good practice?

It's necessary to cast numbers to str if we are appending it with other strings in print statements
ex: print("number" + str(number))  --> no error
     print("number" + number ) --> type error
     print(number) --> no error

File: Factors.py (hunk 1 above)
Comment (Harshitha-Somashekar) [reply Factors.py]: fixed

**Review by git4sudar — APPROVED** [review]

**Pull request merged**